src/Serilog.Sinks.Raygun/Sinks/Raygun/RaygunSink.cs

```diff
@@ -109,12 +113,37 @@ public void Emit(LogEvent logEvent)
                 : new RaygunErrorMessage()
                 {
                     ClassName = logEvent.MessageTemplate.Text,
-                    Message = logEvent.MessageTemplate.Text,     
+                    Message = logEvent.MessageTemplate.Text,
                     Data = logEvent.Properties.ToDictionary(k => k.Key, v => v.Value.ToString())
                 };
 
             // Add user when requested
-            if (!string.IsNullOrWhiteSpace(_userNameProperty) &&
+            if (!string.IsNullOrWhiteSpace(_userInfoProperty) &&
+                logEvent.Properties.ContainsKey(_userInfoProperty) &&
+                logEvent.Properties[_userInfoProperty] != null)
+            {
+                var userInfoStructure = logEvent.Properties[_userInfoProperty] as StructureValue;
+                if (userInfoStructure != null)
```

Comment:
`if (userInfoProperty is StructureValue userInfoStructure)` perhaps?

Reply:
In that case, I've gone an extra step and turned it into a case pattern matching statement if that's ok :)